### DISCUSSION TITLE
Overriding scheduling of FindRelatedProducts

### DISCUSSION THREAD

**Discussion by avivanoff** [wixtoolset/issues#8100] [Questions]
We have the following fragment to override FindRelatedProducts scheduling:
```
<InstallExecuteSequence>
  <AppSearch After="FindRelatedProducts" />
  <FindRelatedProducts Before="LaunchConditions" />
  <RemoveExistingProducts After="InstallValidate" />
</InstallExecuteSequence>
```
It compiles and works just fine in WIX 4, but after upgrading to WIX5 we get the following error:
```
error WIX7009: The action 'FindRelatedProducts' conflicts with a virtual symbol with the same id. To override the virtual symbol (e.g., to reschedule a custom action), use the 'override' access modifier: 'override FindRelatedProducts'. If you didn't intend to override a virtual symbol, use a different id to avoid the conflict.
```
[Documentation](https://wixtoolset.org/docs/fivefour/#virtual-and-overridable-symbols) gives an example of how to override the custom action, but I cannot figure out how to use this approach with FindRelatedProducts.

**Comment by avivanoff** [comment]
Anyone? Seems like I am missing something very obvious.

**Comment by avivanoff** [comment]
@robmen, @barnson, any help?

**Reply by robmen** [reply]
Dude, please, chill out. We've been working really hard on all these releases for WiX the last few months. It's time for us to slow down a little bit and focus on just our customers for a while. None of our customers are reporting this issue, so it's not that important at this time.

Remember, if you're not a customer, the `-rc` process provides the window where we focus on bugs reported against the next major version. If you wait until _after_ the WiX Toolset releases to try things out, then you've missed the window, and you'll need to be patient. The next WiX Online Meeting is on April 16th. It would be great to make sure an issue is opened so it gets triaged.

**Reply by avivanoff** [reply]
Is there an issue? I suspect I just do not understand how to properly use virtual/override.

**Reply by robmen** [reply]
Ahh, I've not looked closely (yet) but I expect this is an issue.

**Reply by campersau** [reply]
I got the same issue with `FindRelatedProducts` and also `AppSearch` / `CostInitialize` after upgrading from WIX 4.0.4 to 5.0.0.

**Comment by robmen** [comment]
By the way, I apologize for [overreacting yesterday](https://github.com/orgs/wixtoolset/discussions/8100#discussioncomment-9061476). I was frustrated that (very likely) a bug found right after releasing WiX v5.

I appreciate you were trying to understand whether this was a new feature (`virtual`/`override`) you needed to understand or a bug. At this point, it looks like a test case is missing and that this is a bug.

**Reply by avivanoff** [reply]
Created a bug wixtoolset/issues#8115.

**Comment by MatthiasRInst** [comment]
This bug has been scheduled for WiX 6 only.

What actually goes wrong is:

If a standard action is explicitly mentioned as a child of a *Sequence element AND it is referred to by another “action scheduling element” in the same sequence, then we get the error WIX7009 at link time.

Here, “referred to”  means mention of the same action in “Before” or “After” attributes of other actions. These references can be in the same module or in a wixlib or even (seems to be a rare case though) within WiX code – in particular for “InstallValidate”, since there is some WiX Code that schedules RemoveExistingProducts after InstallValidate.
In the example above, FindRelatedProducts is a child of the InstallExecuteSequence element and is referred to by AppSearch.

I see only 2 potential workarounds:
1)	Avoid re-scheduling of standard actions.
2)	If re-scheduling or setting a condition is unavoidable, do not refer to the same standard action in other action scheduling elements in the same sequence..
